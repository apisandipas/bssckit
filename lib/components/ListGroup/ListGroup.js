"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styledTools = require("styled-tools");

var _theme = _interopRequireDefault(require("../../theme"));

var _base = require("../Utilities/base");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n  display: flex;\n  flex-direction: column;\n  padding-left: ", ";\n  margin-bottom: ", ";\n  ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n    & > a,\n    & > li {\n      border-right: ", ";\n      border-left: ", ";\n      border-radius: ", ";\n      &:first-child {\n        border-top: ", ";\n      }\n      &:last-child {\n        border-bottom: ", ";\n      }\n    }\n  "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var getBorder = function getBorder(props) {
  return props.flush && (0, _styledComponents.css)(_templateObject(), (0, _styledTools.theme)('listGroup.border.aLiFlushRight'), (0, _styledTools.theme)('listGroup.border.aLiFlushLeft'), (0, _styledTools.theme)('listGroup.borderRadius.noRadius'), (0, _styledTools.theme)('listGroup.border.aLiFlushFirstChildTop'), (0, _styledTools.theme)('listGroup.border.aLiFlushLastChildBottom'));
};

var ListGroup = (0, _styledComponents["default"])(_base.Ul)(_templateObject2(), (0, _styledTools.theme)('listGroup.padding.left'), (0, _styledTools.theme)('listGroup.margin.bottom'), function (props) {
  return getBorder(props);
});
ListGroup.defaultProps = {
  theme: _theme["default"]
};
var _default = ListGroup;
exports["default"] = _default;