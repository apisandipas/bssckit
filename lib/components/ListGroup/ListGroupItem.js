"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _themeFunctions = require("../../utils/themeFunctions");

var _base = require("../Utilities/base");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _templateObject14() {
  var data = _taggedTemplateLiteral(["\n  position: relative;\n  display: block;\n  padding: ", ";\n  border: ", " ", ";\n  border-bottom: none;\n  margin-bottom: -1px;\n  &:first-child {\n    border-top-left-radius: ", ";\n    border-top-right-radius: ", ";\n  };\n  &:last-child {\n    border-bottom: ", "\n    ", ";\n    margin-bottom: 0;\n    border-bottom-right-radius: ", ";\n    border-bottom-left-radius: ", ";\n  };\n  & > a,\n  & > a:hover,\n  & > a:focus {\n    z-index: 1;\n    text-decoration: none;\n  };\n  ", "\n  ", "\n  ", "\n  ", "\n"]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = _taggedTemplateLiteral(["\n        &:hover,\n        &:focus,\n        & > a:hover,\n        & > a:focus {\n          cursor: text;\n        }\n      "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = _taggedTemplateLiteral(["\n        &:hover,\n        &:focus,\n        & > a:hover,\n        & > a:focus {\n          cursor: pointer;\n        }\n      "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = _taggedTemplateLiteral(["\n    z-index: 2;\n  "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = _taggedTemplateLiteral(["\n        background-color: ", ";\n        border-color: ", ";\n        &:hover,\n        &:focus,\n        & > a:hover,\n        & > a:focus {\n          background-color: ", ";\n        }\n      "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = _taggedTemplateLiteral(["\n        &:hover,\n        &:focus,\n        & > a:hover,\n        & > a:focus {\n          background-color: ", ";\n        }\n      "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = _taggedTemplateLiteral(["\n    background-color: ", ";\n    border-color: ", ";\n    ", "\n    ", "\n  "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = _taggedTemplateLiteral(["\n      background-color: ", ";\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = _taggedTemplateLiteral(["\n        & > a,\n        & > a:hover,\n        & > a:focus {\n          color: ", ";\n        }\n      "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = _taggedTemplateLiteral(["\n    color: ", ";\n    ", "\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = _taggedTemplateLiteral(["\n          & > a,\n          & > a:hover,\n          & > a:focus {\n            color: ", ";\n          }\n        "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = _taggedTemplateLiteral(["\n      color: ", ";\n      ", "\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = _taggedTemplateLiteral(["\n          & > a,\n          & > a:hover,\n          & > a:focus {\n            color: ", ";\n          }\n        "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = _taggedTemplateLiteral(["\n      color: ", ";\n      ", "\n    "]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteral(strings, raw) { if (!raw) { raw = strings.slice(0); } return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } })); }

var color = function color(props) {
  if (props.disabled) {
    return (0, _styledComponents.css)(_templateObject(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'colorDisabled'), props.action && (0, _styledComponents.css)(_templateObject2(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'colorDisabled')));
  } else if (props.active) {
    return (0, _styledComponents.css)(_templateObject3(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'colorActive'), props.action && (0, _styledComponents.css)(_templateObject4(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'colorActive')));
  }

  return (0, _styledComponents.css)(_templateObject5(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'color'), props.action && (0, _styledComponents.css)(_templateObject6(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'color')));
};

var backgroundBorderColor = function backgroundBorderColor(props) {
  if (props.disabled) {
    return (0, _styledComponents.css)(_templateObject7(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'backgroundColorDisabled'));
  }

  return (0, _styledComponents.css)(_templateObject8(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'backgroundColor'), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'borderColor'), props.action && (0, _styledComponents.css)(_templateObject9(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'backgroundColorHoverFocusAction')), props.active && (0, _styledComponents.css)(_templateObject10(), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'backgroundColorActive'), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'borderColorActive'), (0, _themeFunctions.getColor)(props, 'listGroupItem', 'borderColorActive')));
};

var zIndex = function zIndex(props) {
  return props.active && (0, _styledComponents.css)(_templateObject11());
};

var cursor = function cursor(props) {
  return !props.disabled && props.action ? (0, _styledComponents.css)(_templateObject12()) : (0, _styledComponents.css)(_templateObject13());
};

var ListGroupItem = (0, _styledComponents["default"])(_base.Li)(_templateObject14(), function (props) {
  return (0, _themeFunctions.getPadding)(props, 'listGroupItem', 'default');
}, function (props) {
  return (0, _themeFunctions.getBorder)(props, 'listGroupItem', 'default');
}, function (props) {
  return (0, _themeFunctions.getColor)(props, 'listGroupItem', 'borderColor');
}, function (props) {
  return (0, _themeFunctions.getBorderRadius)(props, 'listGroupItem', 'default');
}, function (props) {
  return (0, _themeFunctions.getBorderRadius)(props, 'listGroupItem', 'default');
}, function (props) {
  return (0, _themeFunctions.getBorder)(props, 'listGroupItem', 'default');
}, function (props) {
  if (props.active) {
    return (0, _themeFunctions.getColor)(props, 'listGroupItem', 'borderColorActive');
  }

  return (0, _themeFunctions.getColor)(props, 'listGroupItem', 'borderColor');
}, function (props) {
  return (0, _themeFunctions.getBorderRadius)(props, 'listGroupItem', 'default');
}, function (props) {
  return (0, _themeFunctions.getBorderRadius)(props, 'listGroupItem', 'default');
}, function (props) {
  return color(props);
}, function (props) {
  return backgroundBorderColor(props);
}, function (props) {
  return zIndex(props);
}, function (props) {
  return cursor(props);
});
var _default = ListGroupItem;
exports["default"] = _default;