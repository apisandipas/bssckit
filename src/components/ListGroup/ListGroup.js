import styled, { css } from 'styled-components';
import defaultTheme from '../../theme';
import { getPadding, getMargin, getBorder, getBorderRadius } from '../../utils/themeFunctions';
import { Ul } from '../Utilities/base';

const border = props =>
  props.flush &&
  css`
    & > a,
    & > li {
      border-right: ${getBorder(props, 'listGroup', 'aLiFlushRight')};
      border-left: ${getBorder(props, 'listGroup', 'aLiFlushLeft')};
      border-radius: ${getBorderRadius(props, 'listGroup', 'noRadius')};
      &:first-child {
        border-top: ${getBorder(props, 'listGroup', 'aLiFlushFirstChildTop')};
      }
      &:last-child {
        border-bottom: ${getBorder(props, 'listGroup', 'aLiFlushLastChildBottom')};
      }
    }
  `;

const ListGroup = styled(Ul)`
  display: flex;
  flex-direction: column;
  padding-left: ${props => getPadding(props, 'listGroup', 'left')};
  margin-bottom: ${props => getMargin(props, 'listGroup', 'bottom')};
  ${props => border(props)};
`;

ListGroup.defaultProps = {
  theme: defaultTheme,
};

export default ListGroup;
