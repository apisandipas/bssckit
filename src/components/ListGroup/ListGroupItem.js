import styled, { css } from 'styled-components';
import { theme, ifProp } from 'styled-tools';
import { themeContexts } from '../../utils/consts';
import defaultTheme from '../../theme';
import { Li } from '../Utilities/base';

const getColor = (props, context) => {
  if (props.disabled) {
    return css`
      color: ${theme(`listGroupItem.colors.${context}.colorDisabled`)};
      ${props.action &&
        css`
          & > a,
          & > a:hover,
          & > a:focus {
            color: ${theme(`listGroupItem.colors.${context}.colorDisabled`)};
          }
        `}
    `;
  } else if (props.active) {
    return css`
      color: ${theme(`listGroupItem.colors.${context}.colorActive`)};
      ${props.action &&
        css`
          & > a,
          & > a:hover,
          & > a:focus {
            color: ${theme(`listGroupItem.colors.${context}.colorActive`)};
          }
        `}
    `;
  }

  return css`
    color: ${theme(`listGroupItem.colors.${context}.color`)};
    ${props.action &&
      css`
        & > a,
        & > a:hover,
        & > a:focus {
          color: ${theme(`listGroupItem.colors.${context}.color`)};
        }
      `}
  `;
};

const getBackgroundBorderColor = (props, context) => {
  if (props.disabled) {
    return css`
      background-color: ${theme(`listGroupItem.colors.${context}.backgroundColorDisabled`)};
    `;
  }

  return css`
    background-color: ${theme(`listGroupItem.colors.${context}.backgroundColor`)};
    border-color: ${theme(`listGroupItem.colors.${context}.borderColor`)};
    ${props.action &&
      css`
        &:hover,
        &:focus,
        & > a:hover,
        & > a:focus {
          background-color: ${theme(`listGroupItem.colors.${context}.backgroundColorHoverFocusAction`)};
        }
      `}
    ${props.active &&
      css`
        background-color: ${theme(`listGroupItem.colors.${context}.backgroundColorActive`)};
        border-color: ${theme(`listGroupItem.colors.${context}.borderColorActive`)};
        &:hover,
        &:focus,
        & > a:hover,
        & > a:focus {
          background-color: ${theme(`listGroupItem.colors.${context}.borderColorActive`)};
        }
      `}
  `;
};

const getBorderColor = (props, context) => {
  if (props.active) {
    return theme(`listGroupItem.colors.${context}.borderColorActive`);
  }
  return theme(`listGroupItem.colors.${context}.borderColor`);
};

const getZIndex = props =>
  props.active &&
  css`
    z-index: 2;
  `;

const getCursor = props =>
  !props.disabled && props.action
    ? css`
        &:hover,
        &:focus,
        & > a:hover,
        & > a:focus {
          cursor: pointer;
        }
      `
    : css`
        &:hover,
        &:focus,
        & > a:hover,
        & > a:focus {
          cursor: text;
        }
      `;

const ListGroupItem = styled(Li)`
  position: relative;
  display: block;
  padding: ${theme('listGroupItem.padding.default')};
  border: ${theme('listGroupItem.border.default')} ${props => getBorderColor(props, 'default')};
  border-bottom: none;
  margin-bottom: -1px;
  &:first-child {
    border-top-left-radius: ${theme('listGroupItem.borderRadius.default')};
    border-top-right-radius: ${theme('listGroupItem.borderRadius.default')};
  }
  &:last-child {
    border-bottom: ${theme('listGroupItem.border.default')} ${props => getBorderColor(props, 'default')};
    margin-bottom: 0;
    border-bottom-right-radius: ${theme('listGroupItem.borderRadius.default')};
    border-bottom-left-radius: ${theme('listGroupItem.borderRadius.default')};
  }
  & > a,
  & > a:hover,
  & > a:focus {
    z-index: 1;
    text-decoration: none;
  }
  ${props => getColor(props, 'default')};
  ${props => getBackgroundBorderColor(props, 'default')};
  ${props => getZIndex(props)};
  ${props => getCursor(props)};

  ${themeContexts.map(context => {
    return ifProp(
      context,
      css`
        ${props => getColor(props, context)};
        ${props => getBackgroundBorderColor(props, context)};
        border: ${theme('listGroupItem.border.default')} ${props => getBorderColor(props, context)};
        &:last-child {
          border-bottom: ${theme('listGroupItem.border.default')} ${props => getBorderColor(props, context)};
        }
      `
    );
  })}
`;

ListGroupItem.defaultProps = {
  theme: defaultTheme,
};

export default ListGroupItem;
