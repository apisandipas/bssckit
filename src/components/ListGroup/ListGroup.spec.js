import React from 'react';
import 'jest-styled-components';
import { render } from '@testing-library/react';

import ListGroup from './ListGroup';

describe('ListGroup Component', () => {
  it('matches normal ListGroup', () => {
    const { container } = render(<ListGroup />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('display', 'flex');
    expect(container.firstChild).toHaveStyleRule('flex-direction', 'column');
    expect(container.firstChild).toHaveStyleRule('padding-left', '0');
    expect(container.firstChild).toHaveStyleRule('margin-bottom', '0');
  });

  it('matches ListGroup flush', () => {
    const { container } = render(<ListGroup flush />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.firstChild).toHaveStyleRule('border-right', '0', {
      modifier: '& > a',
    });
    expect(container.firstChild).toHaveStyleRule('border-left', '0', {
      modifier: '& > a',
    });
    expect(container.firstChild).toHaveStyleRule('border-radius', '0', {
      modifier: '& > a',
    });
    expect(container.firstChild).toHaveStyleRule('border-right', '0', {
      modifier: '& > li',
    });
    expect(container.firstChild).toHaveStyleRule('border-left', '0', {
      modifier: '& > li',
    });
    expect(container.firstChild).toHaveStyleRule('border-radius', '0', {
      modifier: '& > li',
    });
  });
});
