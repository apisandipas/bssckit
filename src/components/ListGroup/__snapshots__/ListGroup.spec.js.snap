// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListGroup Component matches ListGroup flush 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 0;
  margin-bottom: 0;
}

.c0 > a,
.c0 > li {
  border-right: 0;
  border-left: 0;
  border-radius: 0;
}

.c0 > a:first-child,
.c0 > li:first-child {
  border-top: 0;
}

.c0 > a:last-child,
.c0 > li:last-child {
  border-bottom: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<ul
  class="sc-AxiKw c0"
/>
`;

exports[`ListGroup Component matches normal ListGroup 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 0;
  margin-bottom: 0;
}

@media (min-width:576px) {

}

@media (min-width:768px) {

}

@media (min-width:992px) {

}

@media (min-width:1200px) {

}

<ul
  class="sc-AxiKw c0"
/>
`;
